src/core/ddsi/src/q_entity.c

```diff
@@ -3688,7 +3688,7 @@ void writer_set_alive_may_unlock (struct writer *wr, bool notify)
   {
     if (wr->xqos->liveliness.kind == DDS_LIVELINESS_MANUAL_BY_PARTICIPANT)
       participant_add_wr_lease_locked (wr->c.pp, wr);
-    else if (wr->xqos->liveliness.kind == DDS_LIVELINESS_MANUAL_BY_TOPIC)
+    else if (wr->xqos->liveliness.kind == DDS_LIVELINESS_MANUAL_BY_TOPIC && wr->lease_duration != NULL)
```

Comment:
I did a bit of digging and I came up with the following table:
```
QK xqos.liveliness.kind (A, MbP, MbT)
QD xqos.liveliness.lease_duration
D lease_duration
L lease
RENEW lease to renew

QK    QD    D      L      D is stored where?   L is stored where?      RENEW
A     d     ≠null  undef  pp->ldur_auto_wr     -                       -
A     ∞     null   undef  -                    -                       -
MbP   d     ≠null  ≠null  -                    pp->leaseheap_man       minl_man
MbP   ∞     null   undef  -                    -                       minl_man*
MbT   d     ≠null  ≠null  -                    global lease heap       L
MbT   ∞     null   undef  -                    -                       -

"undef" means: not actually initialised to null, which is a problem for (MbT, ∞) cos it checks for L≠null
checking for D≠null does indeed fix that, so does initializing L

"minl_man*" in RENEW: one would think minl_man can’t be null if (MbP, d ≠ ∞)
```
If I got that right, this additional check for `lease_duration != NULL` is superfluous.

Reply:
You're right, the check for lease duration ≠ infinite makes this check superfluous.